tests/test_distgit/test_generic_distgit.py

```diff
@@ -22,194 +24,286 @@ def test_init(self):
         self.assertIsInstance(self.dg, distgit.DistGitRepo)
 
     def test_init_with_branch_override(self):
-        metadata = mock.Mock()
-        metadata.runtime.branch = "original-branch"
+        metadata = flexmock(runtime=flexmock(branch="original-branch"),
+                            config=flexmock(distgit=flexmock(branch=distgit.Missing)),
```

Comment:
@sosiouxme and/or @adammhaile I'd like to ask if you are aware of a more elegant way of defining nested **properties** with flexmock, because this looks UGLY! :nauseated_face: 

I've seen that they offer a nicer API for nested **method** calls, something like:
```python
flexmock(x).should_receive('foo.bar.baz')
# x.foo().bar().baz()
```
but i couldn't find an equivalent for **properties** :disappointed: any advice ?

Reply:
Not that I'm aware of